commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/http/HttpProviderTestCase.java

```diff
@@ -212,4 +215,36 @@ public void testResolveFolderSlashYesRedirectOn() throws FileSystemException {
         testResolveFolderSlash(ConnectionUri + "/read-tests/", true);
     }
 
+    @Test
+    public void testHttpFileSystemFreeUnusedResources() throws Exception {
+        try (StandardFileSystemManager fileSystemManager = new StandardFileSystemManager()) {
+            fileSystemManager.setConfiguration(StandardFileSystemManager.class.getResource("providers.xml"));
+            // use WeakRef
+            fileSystemManager.setFilesCache(new WeakRefFilesCache());
+            fileSystemManager.init();
+
+            String path = "http://www.w3schools.com/webservices/tempconvert.asmx?action=WSDL";
+            AbstractFileSystem http4FileSystem = getFile(fileSystemManager, path);
+            http4FileSystem.isReleaseable();
+
+            while (!http4FileSystem.isReleaseable()) {
```

Comment:
I would fail the test if a timeout is exceeded, probably with the Timeout annotation. We do not want an infinite loop.